### PR TITLE
feat: Sitemap resource

### PR DESCRIPTION
**Pull request by istarkov**
## First Part 

Here is only new local resource `/$resources/sitemap.xml` support

It can't be implemented as route for now as we have 2 issues:
1. Cloudflare does not allow server fetches from same domain
2. Remix does not have fetch like sveltekit allowing to bypass real http roundtrip when fetching local resources

The idea is to write/use it like any route in case of anything of above will be solved.

Most PR Size is generated new fixture page.
Move SiteData into the resource.

Here is example of rendering this resource using html.

Project to play:
https://sitemap.staging.webstudio.is/builder/cab934af-8615-411a-99c1-5822a98eac74 

Published site:
https://sitemap-saas-t6j12.wstd.work/

image

## Next steps:

- Add `origin` to the system, or even better add `system.hostUrl` with origin, hostname etc
- Add xml element and hardcoded sitemap.xml page

## Steps for reproduction

1. click button
4. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/sdk/src/resources-generator.ts [hunk 1]
```diff
@@ -86,10 +86,31 @@ export const generateResourcesLoader = ({
...
+
+  if (hasResources) {
+    generated += `import { sitemap } from "./[sitemap.xml]";\n`;
+  }
+
   generated += `export const loadResources = async (_props: { system: System }) => {\n`;
   generated += generatedVariables;
   if (hasResources) {
+    generated += `
+    const customFetch: typeof fetch = (input, init) => {
```
Comment (TrySound) [inline packages/sdk/src/resources-generator.ts:99]: Seems like static function, maybe put it into sdk?

Comment (istarkov) [reply packages/sdk/src/resources-generator.ts:99]: yes

Comment (istarkov) [reply packages/sdk/src/resources-generator.ts:99]: but probably here will be better as sitedata etc will be arguments

File: apps/builder/app/builder/features/settings-panel/resource-panel.tsx [hunk 2]
```diff
@@ -445,6 +446,10 @@ export const ResourceForm = forwardRef<
       try {
         new URL(evaluatedValue);
       } catch {
+        if (isLocalResource(evaluatedValue, "sitemap.xml")) {
+          return;
+        }
```
Comment (TrySound) [inline apps/builder/app/builder/features/settings-panel/resource-panel.tsx:451]: could be checked before url parser

File: packages/cli/src/prebuild.ts [hunk 3]
```diff
@@ -682,19 +682,7 @@ export const projectMeta: ProjectMeta =
...
-            .filter(
-              (page) =>
-                executeExpression(page.meta.excludePageFromSearch) !== true
-            )
-            .map((page) => ({
-              path: page.path,
-              lastModified: siteData.build.updatedAt,
-            })),
-        },
+        getStaticSiteMapXml(siteData.pages, siteData.build.updatedAt),
```
Comment (TrySound) [inline packages/cli/src/prebuild.ts:685]: xml? but it is json 😁

File: packages/sdk/src/resources-generator.test.ts [hunk 4]
```diff
@@ -35,30 +31,45 @@ test("generate resources loader", () => {
...
+import { sitemap } from "./[sitemap.xml]";
+export const loadResources = async (_props: { system: System }) => {
+
+    const customFetch: typeof fetch = (input, init) => {
+      if (typeof input !== "string") {
+        return fetch(input, init);
+      }
+
+      if (isLocalResource(input, "sitemap.xml")) {
+        // @todo: dynamic import sitemap ???
```
Comment (TrySound) [inline packages/sdk/src/resources-generator.test.ts:45]: what this about?

**Pull request merged**